assembly/src/parsers/ast/mod.rs

```diff
@@ -302,3 +304,20 @@ fn parse_param<I: core::str::FromStr>(op: &Token, param_idx: usize) -> Result<I,
 
     Ok(result)
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Sorts `ProcMap` map by procedure index (`0`th value element) and returns sorted map as vector
+/// `Vec<index, procedure_name, procedure>`
+fn sort_proc_map(proc_map: &ProcMap) -> Vec<(u16, String, ProcedureAst)> {
```

Comment:
I would probably return an `BTreeMap<u16, ProcedureAst>` from here, and you can build this `BTreeMap` pretty easily.

Also, nit: I'd probably name this: `sort_procs_by_index()`.